tests/test_function.py

```diff
@@ -37,9 +322,75 @@ def test_backward(self):
         self.assertEqual(y2.grad[0], 1)
         self.assertEqual(x.grad[0], 2)
 
-    def test_label(self):
-        self.assertEqual(chainer.Function().label,
-                         'Function')
+
+class TestSplit(unittest.TestCase):
+
+    def setUp(self):
+        self.x = numpy.random.uniform(-1, 1, (3, 2)).astype(numpy.float32)
+        self.g1 = numpy.random.uniform(-1, 1, (3, 2)).astype(numpy.float32)
+        self.g2 = numpy.random.uniform(-1, 1, (3, 2)).astype(numpy.float32)
+
+    def _make_split(self, x):
+        v = chainer.Variable(x)
+        v.rank = 1
+        return chainer.function.Split(v)
+
+    def check_init(self, x):
+        split = self._make_split(x)
+        self.assertEqual(split.rank, 1)
+
+    def test_init_cpu(self):
+        self.check_init(self.x)
+
+    @attr.gpu
+    def test_init_gpu(self):
+        self.check_init(cuda.to_gpu(self.x))
+
+    def check_add_branch(self, x):
+        split = self._make_split(x)
+
+        out = split.add_branch()
+        self.assertIsInstance(out, chainer.Variable)
+        self.assertIs(out.creator, split)
+        self.assertEqual(len(split.outputs), 1)
+
+    def test_add_branch_cpu(self):
+        self.check_add_branch(self.x)
+
+    @attr.gpu
+    def test_add_branch_gpu(self):
+        self.check_add_branch(cuda.to_gpu(self.x))
+
+    def check_backward(self, x, g1, g2):
+        split = self._make_split(x)
+
+        grads = (g1, g2, None)
+        gx, = split.backward((x,), grads)
+        gradient_check.assert_allclose(g1 + g2, gx)
+
+    def test_backward_cpu(self):
+        self.check_backward(self.x, self.g1, self.g2)
+
+    @attr.gpu
+    def test_backward_gpu(self):
+        self.check_backward(cuda.to_gpu(self.x),
+                            cuda.to_gpu(self.g1),
+                            cuda.to_gpu(self.g2))
+
+    def check_backward_one(self, x, g1):
+        split = self._make_split(x)
+
+        grads = (g1,)
+        gx, = split.backward((x,), grads)
+        gradient_check.assert_allclose(g1, gx)
```

Comment:
If there is only one output, Split.backward should return the gradient array without copy.
So how about checking it by self.assertIs instead of assert_allclose?